event-sourcing/event-sourcing-pojo/src/main/scala/org/apache/james/eventsourcing/Event.scala

```diff
@@ -19,11 +19,9 @@
 package org.apache.james.eventsourcing
 
 object Event {
-  def belongsToSameAggregate(events: List[_ <: Event]) = events
-    .view
+  def belongsToSameAggregate(events: List[_ <: Event]): Boolean = events
     .map(event => event.getAggregateId)
-    .distinct
-    .take(2)
+    .toSet
```

Comment:
take (limit in java) was a micro-optimization, no? `toSet` needs to iter though all the results, no?